lib/runner.js

```diff
@@ -261,6 +234,156 @@ module.exports = class Runner {
+		if (Object.keys(total).length === 1) {
+			// Store the key used for this test
+			const testKey = Object.keys(total)[0];
+
+			// If no data exists for the values, use an empty array. That way .length calls return expected results
+			!passes[testKey] && (passes[testKey] = []);
+			!failures[testKey] && (failures[testKey] = []);
+			!slows[testKey] && (slows[testKey] = []);
+			!skipped[testKey] && (skipped[testKey] = []);
+
+			header = this._reportHeader(total, passes, failures, slows, skipped, timeTaken);
+
+			// If the test had failures, then generate the failure list for it
+			if (failures[testKey].length) {
+				failureBody += this._testIssues('Failure', failures, testKey);
+			}
+			if (slows[testKey].length) {
+				failureBody += this._testIssues('Slow', slows, testKey);
+			}
+		} else {
+			// There was more than one test ran, so gather data for them all
+			let totalFail = 0,
+				totalPass = 0,
+				totalSkip = 0,
+				totalSlow = 0;
+
+			// Loop through all the test files and count how many total passes, failures, and skips there were
+			for (let i = 0; i < Object.keys(total).length; i++) {
+				for (let j = 0; j < total[Object.keys(total)[i]].length; j++) {
+					const test = total[Object.keys(total)[i]][j].test;
+					switch (test.state) {
+						case 'passed':
+							totalPass++;
+							test.speed === 'slow' && totalSlow++;
+							break;
+						case 'failed':
+							totalFail++;
+							break;
+						default:
+							totalSkip++;
+							break;
+					}
+				}
+			}
+
+			header = this._reportHeader(total, totalPass, totalFail, totalSlow, totalSkip, timeTaken);
+
+			// Loop through all the tests and generate detailed reports for them all
+			for (let i = 0; i < Object.keys(total).length; i++) {
+				const testKey = Object.keys(total)[i];
+
+				// If no data exists for the values, use an empty array. That way .length calls return expected results
+				!failures[testKey] && (failures[testKey] = []);
+				!slows[testKey] && (slows[testKey] = []);
+
+				if (failures[testKey].length || slows[testKey].length) {
+					failureBody += '‖==============================\n';
+					failureBody += `‖ ${testKey}\n`;
+					failureBody += '‖==============================\n\n';
+				}
+
+				// If the single test had failures, then generate the failure list for it
+				if (failures[testKey].length) {
+					failureBody += this._testIssues('Failure', failures, testKey);
+				}
+				if (slows[testKey].length) {
+					failureBody += this._testIssues('Slow', slows, testKey);
+				}
+
+			}
+		}
+
+		return title + header + failureBody;
+	}
+
+	/**
+	 * Generates a body to be used in an email report that only contains failures and slows
+	 *
+	 * @param {String} suiteName Name of the test suite being ran
+	 * @param {Object} data Data that was generated from the Mocha test run
+	 *
+	 * @returns {String} Title of the report
+	 */
+	_reportConstructor(suiteName, data) {
```

Comment:
This just generates the title, right? this should probably be named `_reportTitle` or an equivalent, rather than constructor